cute_sway_recorder/config_file.py

```diff
@@ -0,0 +1,85 @@
+import configparser
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Optional
+
+DEFAULT_CONFIG_PATH = Path.home() / ".config" / "cute-sway-recorder" / "defaults.conf"
+
+def validate_file_dest(file_dest: Path):
+    """
+    Tests if a file destination coming from the config file
+    is a valid destination.
+    Namely
+     - It is not a directory
+     - It's parent is an existing directory
+    """
+    if len(file_dest.name) == 0:
+        raise ValueError(f"\"{file_dest}\" does not have a name.")
+
+    if file_dest.is_dir():
+        raise ValueError(f"\"{file_dest}\" is a directory.")
+
+    # NOTE: call to resolve is because
+    # "foo/..".parent == "foo"
+    if not file_dest.expanduser().resolve().parent.is_dir():
+        raise ValueError(f"\"{file_dest}\"'s parent directory does not exist.")
+
+@dataclass
+class ConfigFile:
+    """
+    Contains data parsed from the config file 
+    at `~/.config/cute-sway-recording/defaults.conf`
+    Use :func:`ConfigFile.parse` to parse the ConfigFile.
```

Comment:
Damn, I didn't know that Sphinx notation works in docstrings. TIL, nice!

Reply:
i didnt know it was called Sphinx notation :p TIL